torch/_export/constraints.py

```diff
@@ -1,9 +1,31 @@
-from typing import Optional
+from typing import Optional, Callable, Union
 
+import torch
+from torch.types import Number, _complex
+from torch import SymInt, SymFloat
 from torch._dynamo import allow_in_graph
-from torch.fx.experimental.symbolic_shapes import constrain_range
+from torch.fx.experimental.symbolic_shapes import constrain_range_non_symint
 from torch.utils._sympy.value_ranges import ValueRangeError
 
+# `Scalar` type used in native_functions.ymal will be translated to `Union[Number, _complex]`
+# could cause type error during since `SymInt` or `SymFloat` will be used.
+# Here manually specify the type explicitly.
+sym_constrain_range: Callable[
+    [Union[Number, _complex, SymInt, SymFloat], Optional[int], Optional[int]],
+    None,
+] = torch.sym_constrain_range  # type: ignore[assignment]
+
+
+def _constrain_range(symbol, min: Optional[int] = None, max: Optional[int] = None):
+    if not isinstance(symbol, SymInt):
+        # This is needed to check if an input value (as a real numeric value, or
+        # an value derived from numeric input) for exporting is within range or
+        # not.
+        constrain_range_non_symint(symbol, min=min, max=max)
+        return
+
+    sym_constrain_range(symbol, min, max)
```

Comment:
So you're only ever going to call this if you get a `SymInt`?

Reply:
As mentioned above yes. This indeed a little bit strange though but seems like the meta implementation is not triggered when real value is passed in though. I'm not sure if the real c++ implementation is triggered in this case though but can check.